integration/size-test/check-size.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Script that measures the size of a given test file and compares it
+ * with an entry in a file size golden. If the size deviates by certain
+ * amount, the script will fail with a non-zero exit code.
+ */
+
+import {statSync, readFileSync, writeFileSync} from 'fs';
+import chalk from 'chalk';
+
+const [testId, testFileRootpath, isApprove] = process.argv.slice(2);
+const testFilePath = require.resolve(`angular_material/${testFileRootpath}`);
+const goldenFilePath = require.resolve('../../goldens/size-test.json');
+
+const golden = JSON.parse(readFileSync(goldenFilePath, 'utf8'));
+const fileStat = statSync(testFilePath);
+const actualSize = fileStat.size;
+
+// If in approve mode, update the golden to reflect the new size.
+if (isApprove) {
+  golden[testId] = actualSize;
+  writeFileSync(goldenFilePath, JSON.stringify(golden, null, 2));
+  console.info(chalk.green(`Approved golden size for "${testId}"`));
+  process.exit(0);
+}
+
+// If no size has been collected for the test id, report an error.
+if (golden[testId] === undefined) {
+  console.error(`No golden size for "${testId}". Please create a new entry.`);
+  printApproveCommand();
+  process.exit(1);
+}
+
+const expectedSize = Number(golden[testId]);
+const absoluteSizeDiff = Math.abs(actualSize - expectedSize);
+
+// Whether the size deviates by 1% from the expected.
```

Comment:
Rather than making this percentage based, what if it were just 1 kilobyte? Just because e.g. Overlay is big, doesn't mean we should necessarily tolerate larger size fluctuations in that package.

Reply:
@jelbourn Exactly, but that's why we have an absolute byte change check too. The percentage is helpful when a change is smaller than the absolute threshold but is considered large in percentage of the expected size.

This matches the same conditions as in `angular/angular`. The absolute threshold is currently 500 bytes.

Reply:
@josephperrott Moved it to the top-level as constants w/ comments. That should be better, agreed.